app/Limits.cpp

```diff
@@ -17,12 +17,99 @@ Limits::Limits(const std::shared_ptr<const Params>& params)
   : params_(params) {
 }
 
+double Limits::throttleToSpeed(const double throttle) const {
+  double speed_calc;
+  if (throttle < 0) {
+    speed_calc = throttle * params_->velocity_min;
+  }  else {
+    speed_calc = throttle * params_->velocity_max;
+  }
+  return speed_calc;
+}
+
+double Limits::speedToThrottle(const double speed) const {
+  double throttle_calc;
+  if (speed < 0) {
+    throttle_calc = - (speed / params_->velocity_min);
```

Comment:
I think this should be an error case; `speed` should never be negative. Should this function be called `velocityToThrottle`?

Reply:
Will change to a floor at 0 so worst case negative values cause rover to stop.